mne/io/nedf/nedf.py

```diff
@@ -0,0 +1,181 @@
+    if headerxml.findtext('AdditionalChannelStatus', 'OFF') != 'OFF':
+        raise RuntimeError('Unknown additional channel, aborting.')
+
+    n_acc = int(headerxml.findtext('NumberOfChannelsOfAccelerometer', 0))
+    if n_acc:
+        # expect one sample of u16 accelerometer data per block
+        dt.append(('acc', '>u2', (n_acc,)))
+
+    eegset = headerxml.find('EEGSettings')
+    if eegset is None:
+        raise RuntimeError('No EEG channels found')
+    nchantotal = int(eegset.find('TotalNumberOfChannels').text)
+    info['nchan'] = nchantotal
+    info['sfreq'] = int(eegset.find('EEGSamplingRate').text)
+    info['ch_names'] = [e.text for e in eegset.find('EEGMontage')]
+    # expect nchantotal uint24s
+    datadt.append(('eeg', 'B', (nchantotal, 3)))
+
+    info['units'] = eegset.find('EEGUnits')
+
+    if headerxml.find('STIMSettings'):
+        # 2* -> two stim samples per eeg sample
+        datadt.append(('stim', 'B', (2, nchantotal, 3)))
+
+    if 'AdditionalChannelStatus' in headerxml:
+        raise RuntimeError('Unexpected AdditionalChannelStatus')
+
+    if headerxml.findtext('stepDetails/DeviceClass', '') == 'STARSTIM':
+        print('Found Starstim, not sure how to handle this')
+
+    # Trigger data: 4 bytes in newer versions, 1 byte in older versions
+    trigger_type = '>i4' if headerxml.findtext('NEDFversion') else 'B'
+    datadt.append(('trig', trigger_type))
+    # 5 data samples per block
+    dt.append(('data', np.dtype(datadt), (5,)))
+
+    date = headerxml.findtext('StepDetails/StartDate_firstEEGTimestamp', '')
+    info['meas_date'] = datetime.datetime.utcfromtimestamp(int(date) / 1000)
+    return info, np.dtype(dt)
+
+
+def _read_nedf_eeg(filename: str):
+    """
+    Read header info and EEG data from an .nedf file
+
+    Parameters
+    ----------
+    filename : str
+        Path to the .nedf file.
+
+    Returns
+    -------
+    eeg : array, shape (n_samples, n_channels)
+        Unscaled EEG data
+    info : dict
+        Information from the file header
+    triggers : array, shape (n_annots, 2)
+        Start samples and values of each trigger
+    scale : float
+        Scaling factor for the EEG data
+    """
+    info, dt = parse_nedf_header(filename)
+
+    # to quote the original matlab implementation:
+    # "binary data will always start at byte 5120"
+    binstart = 10240
+    with open(filename, mode='rb') as f:
+        f.seek(binstart, os.SEEK_SET)
+        data = np.fromfile(f, dtype=dt)
+
+    # convert uint8-triplet -> float32
+    eeg = data['data']['eeg'] @ [1 << 16, 1 << 8, 1.]
+    eeg = eeg.reshape((-1, info['nchan']))
+    # convert sign if necessary
+    eeg[eeg > (1 << 23)] -= 1 << 24
+
+    triggers = data['data']['trig'].flatten()
+    triggerind = triggers.nonzero()[0]
+    triggers = np.stack((triggerind, triggers[triggerind])).T
+
+    # scale channels accordingly (here: to volts)
+    scale = 2.4 / (6.0 * 8388607)
```

Comment:
What do these numbers mean?

Reply:
I have no idea, but it's the same scaling factor the EEGLAB plugin uses.

Reply:
Who wrote that plugin? The company? Then maybe mention that these magic numbers come from the company itself.

Reply:
Done.